umh.docs.umh.app/content/en/docs/datacontracts/_index.md

```diff
@@ -0,0 +1,213 @@
+[Data Modeling in the UNS](https://learn.umh.app/lesson/data-modeling-in-the-unified-namespace-mqtt-kafka/)
+on our Learn page.
+<!-- Add link to Jeremys upcoming article, once it is released. -->
+{{% /notice %}}
+
+## Components of a Data Contract
+
+Data Contracts have several components and rules. This section provides a
+high-level overview of these key elements. The specifics can vary greatly
+between Data Contracts; therefore, detailed information about the
+[Historian Data Contract](https://umh.docs.umh.app/docs/datacontracts/historian/)
+and [Custom Data Contracts](https://umh.docs.umh.app/docs/datacontracts/customdatacontracts/)
+is provided on their respective pages.
+
+### Topic structure
+
+As mentioned in the example, messages in the UMH must follow a specific ISA-95
+compliant structure in order to be processed. The structure itself can be
+divided into several sections.
+
+{{<mermaid theme="neutral" >}}
+flowchart LR
+    umh --> v1
+    v1 --> enterprise
+    enterprise -->|Optional| site
+    site -->|Optional| area
+    area -->|Optional| productionLine
+    productionLine -->|Optional| workCell
+    workCell -->|Optional| originID
+    originID -->|Optional| _schema["_schema (Ex: _historian, _custom)"]
+    _schema -->_opt["Schema dependent context"]
+
+    classDef mqtt fill:#00dd00,stroke:#333,stroke-width:4px;
+    class umh,v1,enterprise,_schema mqtt;
+    classDef optional fill:#77aa77,stroke:#333,stroke-width:4px;
+    class site,area,productionLine,workCell,originID optional;
+    
+    enterprise -.-> _schema
+    site -.-> _schema
+    area -.-> _schema
+    productionLine -.-> _schema
+    workCell -.-> _schema
+
+    click _schema href "#_schema"
+    click umh href "#versioning-prefix"
+    click v1 href "#versioning-prefix"
+{{</ mermaid >}}
+
+You can check if your topics are correct in the validator below.
+
+{{% topic_decoder %}}
+
+#### Prefix
+
+The first section is the mandatory prefix: `umh.v1.` It ensures that the
+structure can evolve over time without causing confusion or compatibility
+problems.
+
+#### Location
+
+The next section is the Location, which consists of six parts:
+`enterprise.site.area.productionLine.workCell.originID`.
+
+You may be familiar with this structure as it is used by your instances and
+connections. Here the `enterprise` field is mandatory.
+
+{{% notice note %}}
+When you create a Protocol Converter, it uses the Location of the instance and
+the connection to prefill the topic, but you can add the unused ones or change
+the prefilled parts.
+{{% /notice %}}
+
+#### Schemas
+
+After the Location section, you need to specify a schema. In general, the
+schema tells the UMH what to do with incoming messages, what rules to apply to
+them, or can be used for Data Flow Components to process these messages instead
+of others. In short, it specifies the data contract to be used.
+
+Due to its importance, the schema is highlighted with an underscore.
```

Comment:
its not importance, it is to be technically parsable. if we would not have the underscore then it would be hard to identify whether this is part of the location or if this would be the start of the data contract. therefore, location (like an enterprise) are not allowed to start with an underscore

Reply:
But taggroups can start with an _? so I can use `umh.v1.location._historian._tagGroup.tagname` as a topic?

![image](https://github.com/user-attachments/assets/77b86240-5e81-45cc-bc7a-825b8f2572ab)

The topic validator needs some work...